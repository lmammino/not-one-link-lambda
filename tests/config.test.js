'use strict'

const getConfigFromEnv = require('../src/config')

it('should crete a configuration object from env vars', async () => {
  process.env = {
    DB_PATH: 'some-db-path',
    TAG_AE: 'aetag',
    TAG_DE: 'detag',
    WHITELIST: 'https://www.amazon.com/dp/1839214112?tag=loige0e-20;https://www.amazon.com/dp/B01D8HIIFU?tag=loige0e-20;https://www.amazon.com/dp/B08CHMDKW2?tag=loige0e-20'
  }

  const config = getConfigFromEnv()

  expect(config).toMatchSnapshot()
})

it('should create a configuration object with no whitelist if no WHITELIST is given', async () => {
  process.env = {
    DB_PATH: 'some-db-path',
    TAG_AE: 'aetag',
    TAG_DE: 'detag'
  }

  const config = getConfigFromEnv()

  expect(config).toMatchSnapshot()
})

it('should use a default db path if no one is specified', async () => {
  process.env = {}

  const config = getConfigFromEnv()

  expect(config).toMatchSnapshot()
})
