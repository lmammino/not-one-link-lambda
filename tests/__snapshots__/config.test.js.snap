// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a configuration object with no whitelist if no WHITELIST is given 1`] = `
Object {
  "dbPath": "some-db-path",
  "tagsMap": Object {
    "www.amazon.ae": "aetag",
    "www.amazon.ca": undefined,
    "www.amazon.cn": undefined,
    "www.amazon.co.jp": undefined,
    "www.amazon.co.uk": undefined,
    "www.amazon.com": undefined,
    "www.amazon.com.au": undefined,
    "www.amazon.com.br": undefined,
    "www.amazon.com.mx": undefined,
    "www.amazon.com.tr": undefined,
    "www.amazon.de": "detag",
    "www.amazon.es": undefined,
    "www.amazon.fr": undefined,
    "www.amazon.in": undefined,
    "www.amazon.it": undefined,
    "www.amazon.nl": undefined,
    "www.amazon.sg": undefined,
  },
  "whitelist": undefined,
}
`;

exports[`should crete a configuration object from env vars 1`] = `
Object {
  "dbPath": "some-db-path",
  "tagsMap": Object {
    "www.amazon.ae": "aetag",
    "www.amazon.ca": undefined,
    "www.amazon.cn": undefined,
    "www.amazon.co.jp": undefined,
    "www.amazon.co.uk": undefined,
    "www.amazon.com": undefined,
    "www.amazon.com.au": undefined,
    "www.amazon.com.br": undefined,
    "www.amazon.com.mx": undefined,
    "www.amazon.com.tr": undefined,
    "www.amazon.de": "detag",
    "www.amazon.es": undefined,
    "www.amazon.fr": undefined,
    "www.amazon.in": undefined,
    "www.amazon.it": undefined,
    "www.amazon.nl": undefined,
    "www.amazon.sg": undefined,
  },
  "whitelist": Array [
    "https://www.amazon.com/dp/1839214112?tag=loige0e-20",
    "https://www.amazon.com/dp/B01D8HIIFU?tag=loige0e-20",
    "https://www.amazon.com/dp/B08CHMDKW2?tag=loige0e-20",
  ],
}
`;

exports[`should use a default db path if no one is specified 1`] = `
Object {
  "dbPath": "data/GeoLite2-Country.mmdb",
  "tagsMap": Object {
    "www.amazon.ae": undefined,
    "www.amazon.ca": undefined,
    "www.amazon.cn": undefined,
    "www.amazon.co.jp": undefined,
    "www.amazon.co.uk": undefined,
    "www.amazon.com": undefined,
    "www.amazon.com.au": undefined,
    "www.amazon.com.br": undefined,
    "www.amazon.com.mx": undefined,
    "www.amazon.com.tr": undefined,
    "www.amazon.de": undefined,
    "www.amazon.es": undefined,
    "www.amazon.fr": undefined,
    "www.amazon.in": undefined,
    "www.amazon.it": undefined,
    "www.amazon.nl": undefined,
    "www.amazon.sg": undefined,
  },
  "whitelist": undefined,
}
`;
